sodetlib/operations/bias_dets.py

```diff
@@ -201,3 +201,347 @@ def bias_to_rfrac(S, cfg, rfrac=0.5, bias_groups=None, iva=None,
+    transition. Next step is making this script can handle extreme cases faster
+
+
+    """
+
+    ## take the initial biasstep
+    S.log("taking the first biasstep")
+    intical_current_biasvoltage = S.get_tes_bias_bipolar_array()
+    S.log(f"current biasvoltage {intical_current_biasvoltage}")
+    bsa_0 = bias_steps.take_bias_steps(S, cfg)
+    if not math_only:
+        bias_steps.plot_Rfrac(bsa_0)
+            
+    ## load IV analysis result so can use dynamic step size
+    iva = iv.IVAnalysis.load(bsa_0.meta['iv_file'])
+                
+    repeat_biasstep = False
+    ## check for script failue
+    for bl in bias_groups:
+        mask_bg = np.where(bsa_0.bgmap==bl)[0]
+        per_bl_R0 = bsa_0.R0[mask_bg]
+        if np.isnan(np.nanmedian(per_bl_R0)):
+            ## the yield is very low, could it be the biasstep analysis script failed? repeat again
+            repeat_biasstep = True
+            S.log("the biasstep script might have failed, going to repeat the measurement")
+
+    if repeat_biasstep:
+        S.log("retaking biasstep due to failure")
+        bsa_0 = bias_steps.take_bias_steps(S, cfg)
+        repeat_biasstep = False
+    
+    percentage_rn_0 = bsa_0.R0/bsa_0.R_n_IV
+    ## add in checks if detectors are SC  
+    bg_overbias_needed = []
+    overbias_needed = False
+    for bl in bias_groups:
+        mask_bg = np.where(bsa_0.bgmap==bl)[0]
+        per_bl_percentage_rn_0 = percentage_rn_0[mask_bg]
+        ## mask for SC
+        mask_sc = np.where((per_bl_percentage_rn_0 < 0.1) & (per_bl_percentage_rn_0 > -0.1))
+        ## if more than 10 percent of the detectors are SC
+        if len(mask_sc[0]) > 0.1*len(mask_bg):
+            bg_overbias_needed.append(bl)
+            overbias_needed = True
+
+    ## add in check if detectors are normal, hence we should increase the drop step
+    bg_detectors_normal = []
+    drop_from_normal = False
+    for bl in bias_groups:
+        mask_bg = np.where(bsa_0.bgmap==bl)[0]
+        per_bl_percentage_rn_0 = percentage_rn_0[mask_bg]
+        ## mask for normal
+        mask_normal = np.where((per_bl_percentage_rn_0 > 0.9) | (per_bl_percentage_rn_0 < -0.9))
+        ## if more than half of the detectors are normal
+        if len(mask_normal[0]) > 0.5*len(mask_bg):
+            bg_detectors_normal.append(bl)   
+            drop_from_normal = True
+
+
+    if overbias_needed:
+        S.log(f"some BL: {bg_overbias_needed} are stuck in SC, going to over bias them, this takes about 5mins")
+        previous_dc_biases = S.get_tes_bias_bipolar_array()
+        
+        sdl.overbias_dets(S, cfg, bias_groups=bg_overbias_needed)
+        ##force wait time 5 mins for UFM to reach equalibrium 
+        for sleep_index in range(5):
+            time.sleep(60)
+        safe_dc_biases = previous_dc_biases.copy()
+        for replace_bg in bg_overbias_needed:
+            safe_dc_biases[replace_bg] = cfg.dev.bias_groups[replace_bg]["testbed_100mK_bias_voltage"]
```

Comment:
Is that a standard entry in the config file? In other words, is it guaranteed to exist?

Reply:
this is not a standard entry but should be added in.
the purpose for this entry is so the script has a reference of what bias_voltage to use such that the detectors won't be in SC state at the site. This can also be understood as the bias_voltages at the minimum loading.
since all modules are tested in Princeton first, the dark detector bias_voltages are naturally the bias_voltages at the minimum loading. If this info is missing for some reason, user can put in some high numbers such as 15V for Uv31 as a start point.

Reply:
Can you do some combination of (a) adding that parameter to [det_config.py](https://github.com/simonsobs/sodetlib/blob/master/sodetlib/det_config.py) (b) using a `.get()` call here to return a sensible default if no matching entry is found?

Reply:
is there a way in the current det_config.py such that we can load different numbers for 90s and 150s?

Reply:
Could we name this something other than testbed bias voltage? I saw "testbed" and was very confused about why that was useful on-site until I saw you explanation here. 

"dark_bias_voltage" or even "reference_bias_voltage" could make this clearer. And it would (correctly) imply that we could change these numbers later as we start to know the UFMs and on-sky loading better.